modules/search-service-impl/src/main/java/org/opencastproject/search/impl/solr/SolrRequester.java

```diff
@@ -760,13 +770,34 @@ private SolrQuery getForAction(SearchQuery q, String action, boolean applyPermis
       if (roles.size() > 0) {
         sb.append(" AND (");
         StringBuilder roleList = new StringBuilder();
+        var ids = new ArrayList<String>(roles.size());
         for (Role role : roles) {
-          if (roleList.length() > 0) {
-            roleList.append(" OR ");
-          }
-          roleList.append(Schema.OC_ACL_PREFIX).append(action).append(":").append(SolrUtils.clean(role.getName()));
-          if (role.getName().equalsIgnoreCase(securityService.getOrganization().getAnonymousRole())) {
-            userHasAnonymousRole = true;
+          var roleName = role.getName();
```

Comment:
`applyPermissions` can still be `true` if the query did not contain an id , see line 639.

Reply:
Ah, I see, I overlooked that part, thx. In that case, I only think that we need to consider the action parameter here as well, instead of allowing both every time.